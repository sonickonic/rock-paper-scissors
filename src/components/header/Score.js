import React from "react";
import styled from "styled-components";

const Container = styled.div`
  text-align: center;
  background-color: var(--white);
  padding: clamp(1rem, 2vw, 2rem) clamp(2.4rem, 4vw, 5rem);
  border-radius: 0.4rem;
`;

const Title = styled.h3`
  color: var(--score-text);
  font-size: clamp(1rem, 2vw, 1.6rem);
  font-weight: 600;
  letter-spacing: 0.16em;
`;

const Points = styled.div`
  color: var(--text);
  letter-spacing: -0.025em;
  font-size: clamp(4rem, 4vw, 5.5rem);
  font-weight: 700;
  line-height: 1;
`;

const Score = ({ score }) => {
  return (
    <Container>
      <Title>score</Title>
      <Points>{score}</Points>
    </Container>
  );
};

export default Score;
