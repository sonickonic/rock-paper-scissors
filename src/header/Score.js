import React, { useState } from "react";
import styled from "styled-components";

const Container = styled.div`
  text-align: center;
  background-color: var(--white);
  padding: 1rem 2.4rem;
  border-radius: 0.4rem;
`;

const Title = styled.h3`
  color: var(--score-text);
  font-size: 1rem;
  font-weight: 600;
  letter-spacing: 0.16em;
`;

const Points = styled.div`
  color: var(--text);
  letter-spacing: -0.025em;
  font-size: 40px;
  font-weight: 700;
  line-height: 1;
`;

function Score() {
  const [score, serScore] = useState("0");

  return (
    <Container>
      <Title>score</Title>
      <Points>{score}</Points>
    </Container>
  );
}

export default Score;
