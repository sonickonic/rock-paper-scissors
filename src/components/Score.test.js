import React from "react";
import { render } from "@testing-library/react";
import Score from "./header/Score";

describe("Score", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Score />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("shows score value", () => {
    const score = 0;
    const { getByText } = render(<Score score={score} />);

    expect(getByText("0")).toBeTruthy();
  });
});
