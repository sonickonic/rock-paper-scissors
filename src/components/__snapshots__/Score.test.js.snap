// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm cUUMCg"
  >
    <h3
      class="sc-AxirZ fCOrbU"
    >
      score
    </h3>
    <div
      class="sc-AxiKw jcaSpX"
    />
  </div>
</DocumentFragment>
`;
